generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/finish/password-reset-finish.tsx.ejs

```diff
@@ -16,45 +16,33 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 -%>
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { connect } from 'react-redux';
-import { Alert, Col, Row, Button } from 'reactstrap';
+import { Col, Row, Button } from 'reactstrap';
 import { AvForm, AvField } from 'availity-reactstrap-validation';
 import { Translate, translate, getUrlParameter } from 'react-jhipster';
 import { RouteComponentProps } from 'react-router-dom';
 
-import { IRootState } from 'app/shared/reducers';
 import { handlePasswordResetFinish, reset } from '../password-reset.reducer';
 import PasswordStrengthBar from 'app/shared/layout/password/password-strength-bar';
 
-export interface IPasswordResetFinishProps extends DispatchProps, RouteComponentProps<{key: string}>  {}
+export interface IPasswordResetFinishProps extends DispatchProps, RouteComponentProps<{key: string}> {}
 
-export interface IPasswordResetFinishState {
-  password: string;
-  key: string;
-}
+export const PasswordResetFinishPage = (props: IPasswordResetFinishProps) => {
+  const [password, setPassword] = useState('');
+  const [key, setKey] = useState(getUrlParameter('key', props.location.search));
 
-export class PasswordResetFinishPage extends React.Component<IPasswordResetFinishProps, IPasswordResetFinishState> {
-  state: IPasswordResetFinishState = {
-    password: '',
-    key: getUrlParameter('key', this.props.location.search)
-  };
-
-  componentWillUnmount() {
-    this.props.reset();
-  }
+  useEffect(() => {
+    return () => props.reset();
+  }, []);
```

Comment:
could you please change this to `useEffect(() => () => props.reset(), []);
` 
:smile:

Reply:
@chrisdns ok